enhancements/sha-bundle-digest.md

```diff
@@ -0,0 +1,163 @@
+    
+-   Allow user to specify additional images to resolve digests via an environment variable.
+    
+-   Add ability to resolve image digests from image tag dynamically using local docker registry.
+    
+-   OPEN QUESTION: Store digest and tag mappings in file so that user-written code can resolve tags to digest.
+    
+
+  
+
+### Non-Goals
+
+  
+
+-   Migrate existing Operator-SDK-based operators from image tags to image digests.
+    
+-   Identify locations in user code that reference images by tag.
+    
+
+  
+  
+
+## Proposal
+
+-   Add `--image-digests` flag to `make bundle` in order to enable this feature to be run. TODO(ish): Add example command being run with this flag.
+    
+-   During bundle generation, discover all environment variables that begin with `RELATED_IMAGE_` then resolve the tags to digests. These variables should be set in `config/manager/manager.yaml` as suggested in [https://master.sdk.operatorframework.io/docs/best-practices/common-recommendation/#other-common-suggestions](https://master.sdk.operatorframework.io/docs/best-practices/common-recommendation/#other-common-suggestions). 
+    
+
+```
+ - name: RELATED_IMAGE_che_server_secure_exposer_jwt_proxy_image
+
+   value: quay.io/eclipse/che-jwtproxy:0.10.0
+
+- name: RELATED_IMAGE_single_host_gateway
+
+  value: quay.io/eclipse/che--traefik:v2.5.0-eb30f9f09a65cee1fab5ef9c64cb4ec91b800dc3fdd738d62a9d4334f0114683
+```
+
+-   Generate OCP cluster CSV to allow the docker daemon to resolve image tags to digests. 
+    
+
+  
+  
+
+### Risks and Mitigations
+
+  
+
+Since the feature is op-in, it carries the risk of the end-user enabling it without a correct understanding of the feature and its implications. This can be mitigated with clear documentation both on the website and in the CLI help section.
+
+  
+  
+
+## Design Details
+
+  
+
+ -   Add a `--image-digests` flag to the `operator-sdk generate bundle` command that will enable this functionality.
+    
+ -   Add a variable to the generated Makefile called `USE_IMAGE_DIGESTS` that defaults to false and when enabled, it passes the `--use-image-digests` flag to `operator-sdk generate bundle`.
+    
+ -   When `operator-sdk generate-bundle` is called with the `--use-image-digests` flag enabled, it will execute the following logic.
+    
+	- Search the environment that is defined in the manager.yaml fi for all variables that begin with `RELATED_IMAGE_` to    build a list of images that need to have tags resolved to digests    along with the Operator’s main image and rbac-proxy image.
+   
+	   -  For each image on the list, it will use `operator-manifest-tools` to resolve the image tag to a digest
+   
+	   -  When building the cluster CSV, all images that have been identified as needing a digest instead of a tag should have that  digest as an annotation in the CSV. TODO(ish): add before/after    snippet from the CSV to show the difference.
+    
+ 
```

Comment:
Whitespace and some formatting.

```suggestion
## Design Details

 -   Add a `--image-digests` flag to the `operator-sdk generate bundle` command that will enable this functionality.
 -   Add a variable to the generated `Makefile` called `USE_IMAGE_DIGESTS` that defaults to `false` and when enabled, it passes the `--use-image-digests` flag to `operator-sdk generate bundle`.
 -   When `operator-sdk generate-bundle` is called with the `--use-image-digests` flag enabled, it will execute the following logic.
 	1. Search the environment that is defined in the `manager.yaml` file for all variables that begin with `RELATED_IMAGE_` to build a list of images that need to have tags resolved to digests along with the Operator’s main image and rbac-proxy image.
 	 1.  For each image in the list, it will use `operator-manifest-tools` to resolve the image tag to a digest
 	 1.  When building the cluster CSV, all images that have been identified as needing a digest instead of a tag should have that  digest as an annotation in the CSV. TODO(ish): add before/after snippet from the CSV to show the difference.
```